### PR TITLE
Refactor board region container

### LINKED ISSUE #40: Evaluate SOLID, code smells.
Reviewed the entire code and check for violations of SOLID principles and code smells.

### PR DESCRIPTION
**Pull request by eferos93**
While working on UI, I tried to refactor Board and RegionContainer, by moving some responsibilities from the second to the first. They look more balanced to me now and board now is not a bloater anymore. 
Let me know what do you think.

### CODE REVIEW

**Review by demirbilek95 — APPROVED** [review]
I think it seems better

**Review by wilsonjefferson — APPROVED** [review]

**Comment by eferos93** [comment]
Yo, take a final look so then we merge thiz

File: src/test/java/quentin/core/ChainContainerShould.java [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 import java.util.stream.Stream;
 
 import static org.junit.jupiter.api.Assertions.*;
 import static quentin.core.Position.in;
 
-public class ChainShould {
+public class ChainContainerShould {
 
     @Test
     public void touchEdgesIfItIsACompleteChain() {
```
Comment (wilsonjefferson) [inline src/test/java/quentin/core/ChainContainerShould.java]: ChainContainerShould.touch`Edges`IfItIsACompleteChain should be touch`Sides`IfItIsACompleteChain.

File: src/test/java/quentin/core/ChainContainerShould.java [hunk 2]
```diff
@@ -19,15 +19,9 @@ public void touchEdgesIfItIsACompleteChain() {
...
-        BoardSide.setBoardSize(3);
-        listOfBoardSides.forEach(BoardSide::initialiseSide);
-
-        assertTrue(chain.hasACompleteChain(listOfBoardSides));
+        ChainContainer chainContainer = new ChainContainer(3);
+        intersectionStream.forEach(chainContainer::updateChain);
+        assertEquals(Stone.BLACK, chainContainer.getColorWithCompleteChain());
     }
 
     @Test
```
Comment (wilsonjefferson) [inline src/test/java/quentin/core/ChainContainerShould.java:27]: same here: form edges to sides.

**Comment by wilsonjefferson** [comment]
For me it is okay.

**Review by PascAlex — APPROVED** [review]

**Pull request merged**